### PR TITLE
Implement USB de-initialization flow

### LINKED ISSUE #351: SBL will assert while trying to boot next USB option after enabling USB KB support
To reproduce it on APL,  just enable USB KB by changing following in BoardConfig.py
self.CONSOLE_IN_DEVICE_MASK  = 0x00000003

After it is done,  if USB is not the 1st boot option,  an assertion will be seen.

### PR DESCRIPTION
**Pull request by mauricema**
USB sub-system will have host controller scheduling frames on its
own once it is initialized and enabled. Leaving it running while
payload restarting or OS booting could potentially cause memory
corruption since the DMA might still be running on the background
targeting to previously allocated memory. The safer approach is
to stop the USB controller.

It also fixed #351.

### CODE REVIEW

File: BootloaderCommonPkg/Library/UsbInitLib/UsbInitLib.c [hunk 1]
```diff
@@ -37,12 +26,42 @@ RegisterUsbDevice (
   IN PEI_USB_IO_PPI   *UsbIoPpi
   )
 {
-  if (mUsbIoCount < ARRAY_SIZE (mUsbIoArray)) {
-    mUsbIoArray[mUsbIoCount++] = UsbIoPpi;
+  if (mUsbInit.UsbIoCount < ARRAY_SIZE (mUsbInit.UsbIoArray)) {
+    mUsbInit.UsbIoArray[mUsbInit.UsbIoCount++] = UsbIoPpi;
   }
```
Comment (myguitar) [inline BootloaderCommonPkg/Library/UsbInitLib/UsbInitLib.c]: Better to handle error condition as well.

File: PayloadPkg/OsLoader/OsLoader.c [hunk 2]
```diff
@@ -552,6 +552,9 @@ BeforeOSJump (
   }
```
Comment (myguitar) [inline PayloadPkg/OsLoader/OsLoader.c]: pls put Payload changes in a separate commit.

File: BootloaderCommonPkg/Library/UsbInitLib/UsbInitLib.c (hunk 1 above)
Comment (mauricema) [reply BootloaderCommonPkg/Library/UsbInitLib/UsbInitLib.c]: sure, will add it.

File: PayloadPkg/OsLoader/OsLoader.c (hunk 2 above)
Comment (mauricema) [reply PayloadPkg/OsLoader/OsLoader.c]: will do it.

**Comment by mauricema** [comment]
pushed a new commit per comments.
